adapters/repos/db/lsmkv/bucket.go

```diff
@@ -45,6 +46,9 @@ type Bucket struct {
 	legacyMapSortingBeforeCompaction bool
 
 	stopFlushCycle chan struct{}
+
+	status     storagestate.Status
+	statusLock *sync.Mutex
```

Comment:
is there a reason why `sync.Mutex` is a pointer?

Reply:
I took the example from [the attached internal issue video](https://semi-technology.atlassian.net/browse/WEAVIATE-4), which declared this field as a pointer. I don't have a reason aside from that 🙂 can change to value if preferred